sybilMTP.go

```diff
@@ -0,0 +1,213 @@
+}
+
+func (s SybilMTPInputs) Validate() error {
+
+	if s.ID == nil {
+		return errors.New(ErrorEmptyID)
+	}
+
+	if s.GISTProof.Proof == nil {
+		return errors.New(ErrorEmptyGlobalProof)
+	}
+
+	if s.IssuerClaim.Claim == nil {
+		return errors.New(ErrorEmptyGlobalProof)
+	}
+
+	if s.HolderClaim.Claim == nil {
+		return errors.New(ErrorEmptyGlobalProof)
+	}
+
+	return nil
+}
+
+func (s SybilMTPInputs) InputsMarshal() ([]byte, error) {
+	if err := s.Validate(); err != nil {
+		return nil, err
+	}
+
+	mtpInputs := sybilMTPCircuitInputs{
+		IssuerClaim:           s.IssuerClaim.Claim,
+		IssuerClaimMtp:        CircomSiblings(s.IssuerClaim.IncProof.Proof, s.GetMTLevel()-1),
+		IssuerClaimClaimsRoot: s.IssuerClaim.IncProof.TreeState.ClaimsRoot,
+		IssuerClaimRevRoot:    s.IssuerClaim.IncProof.TreeState.RevocationRoot,
+		IssuerClaimRootsRoot:  s.IssuerClaim.IncProof.TreeState.RootOfRoots,
+		IssuerClaimIdenState:  s.IssuerClaim.IncProof.TreeState.State,
+
+		IssuerClaimNonRevMtp: CircomSiblings(s.IssuerClaim.NonRevProof.Proof, s.GetMTLevel()),
+
+		IssuerClaimNonRevClaimsRoot: s.IssuerClaim.NonRevProof.TreeState.ClaimsRoot,
+		IssuerClaimNonRevRevRoot:    s.IssuerClaim.NonRevProof.TreeState.RevocationRoot,
+		IssuerClaimNonRevRootsRoot:  s.IssuerClaim.NonRevProof.TreeState.RootOfRoots,
+		IssuerClaimNonRevState:      s.IssuerClaim.NonRevProof.TreeState.State,
+
+		// claim of state-secret (Holder's claim)
+		HolderClaim:           s.HolderClaim.Claim,
+		HolderClaimMtp:        CircomSiblings(s.HolderClaim.IncProof.Proof, s.GetMTLevel()-1),
+		HolderClaimClaimsRoot: s.HolderClaim.IncProof.TreeState.ClaimsRoot,
+		HolderClaimRevRoot:    s.HolderClaim.IncProof.TreeState.ClaimsRoot,
+		HolderClaimRootsRoot:  s.HolderClaim.IncProof.TreeState.ClaimsRoot,
+		HolderClaimIdenState:  s.HolderClaim.IncProof.TreeState.ClaimsRoot,
+
+		GistRoot: s.GISTProof.Root,
+		GistMtp:  CircomSiblings(s.GISTProof.Proof, s.GetMTLevel()),
+
+		CRS: s.CRS,
+
+		// user data
+		UserGenesisID:            s.ID.String(),
+		ProfileNonce:             s.ProfileNonce.String(),
+		ClaimSubjectProfileNonce: s.ClaimSubjectProfileNonce.String(),
+	}
+	issuerClaimAux := GetNodeAuxValue(s.IssuerClaim.NonRevProof.Proof)
+	mtpInputs.IssuerClaimNonRevMtpNoAux = issuerClaimAux.noAux
+	mtpInputs.IssuerClaimNonRevMtpAuxHi = issuerClaimAux.key
+	mtpInputs.IssuerClaimNonRevMtpAuxHv = issuerClaimAux.value
+
+	gistAux := GetNodeAuxValue(s.GISTProof.Proof)
+	mtpInputs.GistMtpAuxHi = gistAux.key
+	mtpInputs.GistMtpAuxHv = gistAux.value
+	mtpInputs.GistMtpNoAux = gistAux.noAux
+
+	mtpInputs.RequestID = s.RequestID
+	mtpInputs.IssuerID = s.IssuerID
+	mtpInputs.CurrentTimestamp = s.CurrentTimestamp
+
+	return json.Marshal(mtpInputs)
+}
+
+type SybilMTPPubSignals struct {
+	BaseConfig
+
+	SybilID string `json:"sybilID"`
+	UserID  string `json:"userID"`
+
+	RequestID        string `json:"requestID"`
+	IssuerID         string `json:"issuerID"`
+	CurrentTimestamp string `json:"currentTimestamp"`
```

Comment:
Rename `CurrentTimestamp` to `Timestamp` and change type to int64. It should be in sync with other circuits.